charts/demo-app/templates/deployment.yaml

```diff
@@ -31,7 +31,7 @@ spec:
         - name: {{ .Chart.Name }}
           securityContext:
             {{- toYaml .Values.securityContext | nindent 12 }}
-          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          image: "{{ .Values.image.repository }}@sha256:{{ .Values.image.tag }}"
```

Comment:
Technically what we've been passing as `image.tag` from the `demo-app` has never been a tag, but instead the SHA 256 digest. 

I made the change here to deploy that instead, although the fact that it's called a tag is still misleading. We can update the `demo-app` pipeline to pass both the tag and the digest if we want to fix that.